lib/cocoapods/generator/embed_frameworks_script.rb

```diff
@@ -134,51 +152,34 @@ def script
     eval "$code_sign_cmd"
   fi
 }
-
-install_artifact() {
-  artifact="$1"
-  base="$(basename "$artifact")"
-  case $base in
-  *.framework)
-    install_framework "$artifact"
-    ;;
-  *.dSYM)
-    # Suppress arch warnings since XCFrameworks will include many dSYM files
-    install_dsym "$artifact" "false"
-    ;;
-  *.bcsymbolmap)
-    install_bcsymbolmap "$artifact"
-    ;;
-  *)
-    echo "error: Unrecognized artifact "$artifact""
-    ;;
-  esac
-}
-
-copy_artifacts() {
-  file_list="$1"
-  while read artifact; do
-    install_artifact "$artifact"
-  done <$file_list
-}
-
-ARTIFACT_LIST_FILE="${BUILT_PRODUCTS_DIR}/cocoapods-artifacts-${CONFIGURATION}.txt"
-if [ -r "${ARTIFACT_LIST_FILE}" ]; then
-  copy_artifacts "${ARTIFACT_LIST_FILE}"
-fi
-
         SH
-        frameworks_by_config.each do |config, frameworks_with_dsyms|
-          next if frameworks_with_dsyms.empty?
-          script << %(if [[ "$CONFIGURATION" == "#{config}" ]]; then\n)
-          frameworks_with_dsyms.each do |framework_with_dsym|
-            script << %(  install_framework "#{framework_with_dsym.source_path}"\n)
-            unless framework_with_dsym.bcsymbolmap_paths.nil?
-              framework_with_dsym.bcsymbolmap_paths.each do |bcsymbolmap_path|
-                script << %(  install_bcsymbolmap "#{bcsymbolmap_path}"\n)
+        contents_by_config = Hash.new do |hash, key|
+          hash[key] = ''
+        end
+        frameworks_by_config.each do |config, frameworks|
+          next if frameworks.empty?
+          frameworks.each do |framework|
+            contents_by_config[config] << %(  install_framework "#{framework.source_path}"\n)
+            unless framework.bcsymbolmap_paths.nil?
+              framework.bcsymbolmap_paths.each do |bcsymbolmap_path|
+                contents_by_config[config] << %(  install_bcsymbolmap "#{bcsymbolmap_path}"\n)
               end
             end
           end
+        end
+        xcframeworks_by_config.each do |config, xcframeworks|
+          next if xcframeworks.empty?
```

Comment:
is this really check needed? an empty array below will never loop although this is a bit of a nit.

Reply:
it's not, you're right